examples/bitcoin_crate/src/main.rs

```diff
@@ -28,177 +27,180 @@ struct TxInJson {
             value: tx.output.iter().map(|op| op.value.to_sat()).sum(),
             weight: tx.total_size() as u32,
             input_count: tx.input.len(),
-            creation_sequence: Some(creation_sequence),
+            creation_sequence: Some(i as u32),
         })
-    }
+        .collect()
+}
 
-    Ok(output_group_vec)
+fn calculate_base_weight(inputs: &[TxIn]) -> u32 {
+    println!("Base weights for each UTXO Input: ");
+    let max_weight: u32 = inputs
+        .iter()
+        .map(|input| {
+            let input_size = serialize(input).len();
+            let witness_size = input.witness.iter().map(|w| w.len()).sum::<usize>();
+            let weight = (input_size * 3 + witness_size) as u32;
+            println!(
+                "  Input Size: {}, Witness Size: {}, Weight: {}",
+                input_size, witness_size, weight
+            );
+            weight
+        })
+        .max()
+        .unwrap_or(600);
+    max_weight
 }
 
-fn create_select_options() -> Result<Vec<CoinSelectionOpt>, Box<dyn std::error::Error>> {
-    let mut rng = rand::thread_rng();
-    let mut coin_select_options_vec: Vec<CoinSelectionOpt> = Vec::new();
-    // Creating 5 different options for coin selection
-    for _ in 0..5 {
-        // Random selection of Excess Strategy
-        let excess_strategy = match rng.gen_range(0..3) {
-            0 => ExcessStrategy::ToChange,
-            1 => ExcessStrategy::ToFee,
-            2 => ExcessStrategy::ToRecipient,
-            _ => unreachable!(),
-        };
-        coin_select_options_vec.push(CoinSelectionOpt {
-            target_value: rng.gen_range(40000..5000000000i64) as u64,
-            target_feerate: rng.gen_range(1.0..5.0) as f32,
-            long_term_feerate: Some(rng.gen_range(1..10) as f32),
-            min_absolute_fee: rng.gen_range(1..20) as u64,
-            base_weight: rng.gen_range(1..30) as u32,
-            change_weight: rng.gen_range(5..30) as u32,
-            change_cost: rng.gen_range(1..20) as u64,
-            avg_input_weight: rng.gen_range(1..10) as u32,
-            avg_output_weight: rng.gen_range(1..10) as u32,
-            min_change_value: rng.gen_range(100..1000) as u64,
-            excess_strategy,
+fn create_select_options(inputs: &[TxIn]) -> Vec<CoinSelectionOpt> {
+    let target_values = [50_000, 100_000, 200_000, 500_000, 1_000_000];
+    let feerates = [1.0, 2.0, 3.0, 5.0, 10.0];
+    // Weight defaults to 600 if not provided
+    let base_weight = calculate_base_weight(inputs);
+
+    (0..5)
+        .map(|i| {
+            let excess_strategy = match i % 3 {
+                0 => ExcessStrategy::ToChange,
+                1 => ExcessStrategy::ToFee,
+                2 => ExcessStrategy::ToRecipient,
+                _ => unreachable!(),
+            };
+
+            CoinSelectionOpt {
+                target_value: target_values[i],
+                target_feerate: feerates[i],
+                long_term_feerate: Some(feerates[i]),
```

Comment:
I think we should also be using the 10 sats/vbyte value that core uses as the default longterm fee rate. 
https://github.com/bitcoin/bitcoin/pull/22009

Reply:
Interesting. Thanks for this. I will make this change